viki-kaho/src/Comment.js

```diff
@@ -1,14 +1,85 @@
-function Comment(props) {
-	return (
-		<tr>
-			<td>{props.name}</td>
-			<td>{props.email}</td>
-			<td>{props.body}</td>
-			<td>
-				<button onClick={() => props.handleDelete(props.id)}>Delete</button>
-			</td>
-		</tr>
-	);
+import React from 'react';
+
+class Comment extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			editComment: false,
+			name: this.props.name,
+			email: this.props.email,
+			body: this.props.body,
+		};
+	}
+
+	handleSave = (e) => {
+		e.preventDefault();
+		this.setState({
+			editComment: false,
+		});
+	};
+
+	render() {
+		return this.state.editComment ? (
+			<tr>
+				<td className="hidden">
+					<form id={`form${this.props.id}`} onSubmit={this.handleSave} />
+				</td>
+				<td>
+					<input
+						form={`form${this.props.id}`}
+						type="text"
+						name="name"
+						value={this.state.name}
+						onChange={(e) => this.setState({ name: e.target.value })}
+					/>
+				</td>
+				<td>
+					<input
+						form={`form${this.props.id}`}
+						type="email"
+						name="email"
+						value={this.state.email}
+						onChange={(e) => this.setState({ email: e.target.value })}
+					/>
+				</td>
+				<td>
+					<input
+						form={`form${this.props.id}`}
+						type="text"
+						name="body"
+						value={this.state.body}
+						onChange={(e) => this.setState({ body: e.target.value })}
+					/>
+				</td>
+				<td className="buttons">
+					<button className="save" form={`form${this.props.id}`} type="submit">
```

Comment:
I am not sure if type="submit" is really needed, but we can merge.

Reply:
It is associated with the form and even if it doesn't have `type="submit"` this will act like it has `type="submit"`  by default.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

I want you to double-check it won't cause an error, but I believe it is okay whether it has `type` attribute set explicitly.